File: 530.Minimum.Absolute.Difference.in.BST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function(root) {
    function cmp(a, b){
        return a-b;
    }
    function travel(node, list){
        if (node === null) return;
        list.push(node.val);
        travel(node.left, list);
        travel(node.right, list);
    }
    list = [];
    travel(root, list);
    list = list.sort(cmp);
    ret = list[1] - list[0];
    for(var i = 2; i < list.length; ++i)
    {
        if (list[i] - list[i-1] < ret)
        {
            ret = list[i] - list[i-1];
        }
    }
    return ret;
};

File: 530. Minimum Absolute Difference in BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        def inOrder(node):
            if node.left:
                yield from inOrder(node.left)
            yield node.val
            if node.right:
                yield from inOrder(node.right)
        
        pre = -1
        ret = float('inf')
        for val in inOrder(root):
            if pre !=-1:
                ret = min(ret, abs(val - pre))
            pre = val
        return ret


